apps/gdalalg_vector_geom_simplify_coverage.cpp

```diff
@@ -0,0 +1,352 @@
+/******************************************************************************
+*
+ * Project:  GDAL
+ * Purpose:  "gdal vector geom simplify-coverage" subcommand
+ * Author:   Daniel Baston
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, ISciences LLC
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_vector_geom_simplify_coverage.h"
+
+#include "cpl_error.h"
+#include "gdal_priv.h"
+#include "gdalalg_vector_geom.h"
+#include "ogr_geometry.h"
+#include "ogr_geos.h"
+#include "ogrsf_frmts.h"
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+//! @cond Doxygen_Suppress
+
+GDALVectorGeomSimplifyCoverageAlgorithm::
+    GDALVectorGeomSimplifyCoverageAlgorithm(bool standaloneStep)
+    : GDALVectorPipelineStepAlgorithm(NAME, DESCRIPTION, HELP_URL,
+                                      standaloneStep)
+{
+    AddActiveLayerArg(&m_activeLayer);
+    AddArg("tolerance", 0, _("Distance tolerance for simplification."),
+           &m_opts.tolerance)
+        .SetRequired()
+        .SetMinValueIncluded(0);
+    AddArg("preserve-boundary", 0,
+           _("Whether the exterior boundary should be preserved."),
+           &m_opts.preserveBoundary);
+}
+
+#if defined HAVE_GEOS &&                                                       \
+    (GEOS_VERSION_MAJOR > 3 ||                                                 \
+     (GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR >= 12))
+
+/** Class used by vector pipeline steps to create an output on-the-fly
+ * dataset where they can store on-the-fly layers.
+ */
+class GDALVectorGeomSimplifyCoverageOutputDataset final : public GDALDataset
+{
+  public:
+    GDALVectorGeomSimplifyCoverageOutputDataset(GDALDataset &srcDS);
+
+    void AddLayer(std::unique_ptr<OGRLayer> poNewLayer)
+    {
+        m_layers.push_back(std::move(poNewLayer));
+    }
+
+    int GetLayerCount() override
+    {
+        return static_cast<int>(m_layers.size());
+    }
+
+    OGRLayer *GetLayer(int idx) override
+    {
+        return m_layers[idx].get();
+    }
+
+    int TestCapability(const char *) override
+    {
+        // FIXME implement?
+        return false;
+    }
+
+    void ResetReading() override
+    {
+        m_srcDS.ResetReading();
+    }
+
+    OGRFeature *GetNextFeature(OGRLayer **, double *, GDALProgressFunc,
+                               void *) override
+    {
+        // FIXME implement?
+        CPLError(CE_Failure, CPLE_AppDefined,
+                 "GDALDataset::GetNextFeature not implemented.");
+        return nullptr;
+    }
```

Comment:
The base implementation of GDALDataset will do the right thing:

```suggestion
```

Reply:
removed